Source/WebCore/html/FTPDirectoryDocument.cpp

```diff
@@ -67,14 +66,12 @@ class FTPDirectoryDocumentParser final : public HTMLDocumentParser {
     void append(RefPtr<StringImpl>&&) override;
     void finish() override;
 
-    void checkBuffer(int len = 10)
+    void checkBuffer(size_t len = 10)
     {
-        if ((m_dest - m_buffer) > m_size - len) {
+        if (m_destIndex > (m_size - len)) {
             // Enlarge buffer
             int newSize = std::max(m_size * 2, m_size + len);
```

Comment:
I wonder what guarantees this doesn’t overflow. Seems like this should just use something like `Vector<UChar>`; it’s such old code and so strange that it does its own allocation like this.

Reply:
Ok, I switched to a Vector.

Reply:
Since vectors have capacity and their own rules about allocating capacity as the vector gets larger, we can get rid of the logic about enlarging the buffer, and just resize it to the amount needed. This would eliminate all the `CheckedSize` stuff. I might come back and do that.